extensions/ql-vscode/package.json

```diff
@@ -209,6 +209,24 @@
           "default": null,
           "description": "Path to a directory where the CodeQL extension should store query server logs. If empty, the extension stores logs in a temporary workspace folder and deletes the contents after each run."
         },
+        "codeQL.runningQueries.structuredEvaluatorLog": {
+          "type": [
+            "string",
+            null
+          ],
+          "default": "structured-evaluator-log.json",
+          "description": "Path to a file where the evaluator should store structured evaluator logs. If empty, the query server stores the logs in a default location and overwrites them after each query server run."
+        },
```

Comment:
What is the path relative to? If it must be an absolute path, then we should ensure this is only settable in local workspace settings.

Reply:
Hm, like you mentioned in another comment, there doesn't seem to be a need for us to allow the user to pass this in (as the final state should be that the user views the log via our extension and not as a file itself). 

I can change this to a boolean — just so that users have the ability to have structured logs at all. Or should we enable them by default so that they would only have to toggle the "show" configuration?

Reply:
Is it an either/or? So, users can't produce both normal and structured logs? My thinking is that unless you're trying to process the logs programmatically, non-structured logs would be preferable?

So, we may want to have a boolean for a user to choose.

Reply:
It's not an either-or on the query server side: the `--log-file` argument to the query server is still in effect and both can be produced (although I guess there should be some sort of check in place to make sure the file names passed in aren't the same!) 

I imagined having the "traditional" logs going to the existing logging directory and the structured evaluator logs also going to the same directory, but with a different name, if both are enabled.

Reply:
Presumably, users want the structured logs so they can do things with it. Is this an important use case that we need to support right now?

Also, in the future, the extension itself will be using the logs to provide interesting data directly inline.

I'm fine with getting something out there that is in the general direction we want to go, and then iterating until we have all the bits of the feature in place. I think it's worth having a bit of a think of the use cases so we avoid implementing things we don't really need.

Reply:
The updated commits no longer allow the user to configure a custom location for the logfile. In the future we may want to allow a custom additional location the way the other logfiles do, though.